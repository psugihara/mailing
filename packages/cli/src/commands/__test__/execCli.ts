import { exec } from "child_process";

export async function execCli(command: string, opts?: { debug: boolean }) {
  return new Promise((resolve, reject) => {
    const child = exec(
      `cd packages/cli && FORCE_COLOR=0 ${__dirname}/../../dev.js ${command}`
    );
    let out = "";
    let err = "";
    child.stdout?.on("data", (stream) => {
      out += stream.toString();
      if (opts?.debug) console.log(out);
    });
    child.stderr?.on("data", (stream) => {
      err += stream.toString();
      if (opts?.debug) console.log(out);
    });
    child.on("error", reject);
    child.on("exit", (code) => {
      if (code) {
        if (opts?.debug) {
          console.log(out);
          console.error(err);
        }
        reject(new Error(err));
      }
      resolve(out);
    });
  });
}

export function execCliChild(command: string, _opts?: { debug: boolean }) {
  return exec(`FORCE_COLOR=0 ${__dirname}/../../dev.js ${command}`);
}
