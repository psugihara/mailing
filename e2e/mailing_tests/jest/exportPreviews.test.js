import execCli from "./util/execCli";

describe("exportPreviews command", () => {
  describe("cli", () => {
    it("runs on templates", async () => {
      const out = await execCli("export-previews --skip-lint");
      expect(out).toMatchSnapshot();
    });
  });

  describe("cli --minify", () => {
    it("runs on templates", async () => {
      const out = await execCli("export-previews --minify --skip-lint");
      expect(out).toMatchSnapshot();
    });
  });

  describe("cli halts on lint errors", () => {
    it("runs on templates", async () => {
      const out = await execCli("export-previews");
      expect(out).toMatchSnapshot();
    });
  });
});
