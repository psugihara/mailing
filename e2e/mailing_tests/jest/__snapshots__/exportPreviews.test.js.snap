// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exportPreviews command cli --minify runs on templates 1`] = `
"mailing Exporting minified preview html to
mailing ./previews_html/
mailing   |-- welcome_preview.html
mailing ✅ Processed 1 previews

"
`;

exports[`exportPreviews command cli halts on lint errors runs on templates 1`] = `
"mailing Exporting preview html to
mailing ./previews_html/
mailing   |-- welcome_preview.html
mailing ✅ Processed 1 previews

"
`;

exports[`exportPreviews command cli runs on templates 1`] = `
"mailing Exporting preview html to
mailing ./previews_html/
mailing   |-- welcome_preview.html
mailing ✅ Processed 1 previews

"
`;
